docs/cli.md

```diff
@@ -506,6 +507,26 @@ gcli tag update --tag tagA --rename newTag
 gcli tag update --tag tagA --comment "new comment"
 ```
 
+### Owners commands
+
+#### List an owner
+
+```bash
+gcli catalog details --name postgres --owner
+```
+
+#### Set an owner to a user
+
+```bash
+gcli catalog set --name postgres --user admin
```

Comment:
"set" is better than "update" I think; but as there is no "owner" in any command or parameter, this line is not that readable for user. As owner is such a common concept at different levels (metalake, catalog, schema, table, fileset, etc), I think adding a "set-owner" command would be better to understand and remember. This is my two cents, I think you can request more inputs from others as well.

Reply:
`set-owner` has the same issues as `set`, plus it's inconsistent with other commands. Did you look at the code?

Reply:
I would prefer to use multiple "set-" commands (e.g, set-tag, set-owner) instead of using one "set" command for multiple features (properties, tags, owner etc). The logics in "set" is too complicated to maintain as well as to understand for a user. @jerryshao @xunliu Jerry, Xun, what's your idea about this?

Reply:
@jerryshao @xunliu the currently state of this PR was deliberately done to show why `set` was a poor choice over a previous option - please compare with using a `--owner` flag.

Reply:
I did to show @shaofengshi why the previous solution was better, and it was easier to show that to him in code than in English.

Reply:
My preferred solution is at 315599a8fae3011e83b3bcc0631ed1bd1363b6e3

Reply:
The two options from a user's point of view look like this:
```
gcli catalog details --name postgres --owner
gcli catalog verb --name postgres --user admin --owner
gcli catalog verb --name postgres --group groupA --owner
```
OR
```
gcli catalog details --name postgres --owner
gcli catalog verb --name postgres --user admin
gcli catalog verb --name postgres --group groupA
```

The first option requires 20% of the code of the second option and is much more maintainable. For the second option, we also need to add code when a new entity is added, while the first option will work with no extra code when new entities are added.

I had first suggested using `update` rather than `set` as the verb to use as if you use `set`, then the user has an expectation that `remove` will also work, but it will not.

Reply:
I also feel that `gcli catalog set --name foo --user bar` may not be easy to understand. 

For the reviewers or users, it is not easy to understand that this `set` refers `set owner` unless he knows that `--user` is related to the owner. But if we change to `set-owner`, then users can understand the command easier.

Besides, if later on there's another `set` command also requires `--user` option, how do we differentiate it is a `set-owner` command or anything else?

Reply:
Best practice on develping CLIs has this to say: (https://clig.dev/#subcommands)
"Use consistent names for multiple levels of subcommand. If a complex piece of software has lots of objects and operations that can be performed on those objects, it is a common pattern to use two levels of subcommand for this, where one is a noun and one is a verb. For example, docker container create. Be consistent with the verbs you use across different types of objects."

Reply:
@jerryshao I assume that means you prefer the first options with the `--owner` flags? Using a `set-owner` argument has the same downsides maintenance/code complexity-wise as using just `set` without a `--owner` flag.

Reply:
Re `gcli catalog set --name foo --user bar`, originally I had wanted to use `gcli catalog set --name foo --owner bar` as that is easier to understand. However, there is no easy way to distinguish between a user owner and a group owner. You could get around this by adding three flags, `--owner`, `--user-owner` and `--group-owner`, but that also didn't seem a good solution. I also considered using `chown` style syntax but thought that would be more confusing, especially as you can only have a user or group owner, not both.

Reply:
I would say that comparing these two options, the first one seems more explainable, but do we have better solutions fixing the place where `--owner` should be placed, if we can place `set` and `--owner` closer, then it would be better for users to understand. Or even like @shaofengshi mentioned, using `set-owner`. like:

```
gcli catalog set --owner --name foo --user bar
gcli catalog set --tags xxx
gcli catalog set-owner --name foo --user bar
gcli catalog set-tags xxx
```

Reply:
Flags can occur anywhere, so `gcli catalogue set --owner --name for --user bar` is the same as `gcli catalogue set --name for --user bar --owner`, and we can suggest to the owner to use that order.

Using `gcli catalog set --owner --name foo --user bar` is simpler and more maintainable than `gcli catalog set-owner --name foo --user bar`

Reply:
`gcli catalog set --owner --user bar --name foo` might be the best order to suggest?

Reply:
Yes, I think so.

Reply:
It also applies to other set, like set tags, etc.

Reply:
> Using `set-owner` would not align with that advice.

I don't know why `set-owner` doesn't align with that advice; you mean it is not a "verb"?
But if you look at aws command lines, there are much such sub-commands, for example: 
`aws ec2 describe-instances`, `aws ec2 start-instances --instance-ids i-1348636c`, you can see more such commands at https://docs.aws.amazon.com/cli/latest/reference/glue/

Reply:
> `gcli catalog set --owner --user bar --name foo` might be the best order to suggest?

I won't consider this from how the code is complex, I just from end user's point. In my opinion, "set-owner", "set-tag" is better and clear than "set --owner" and "set --tag". More sub-commands would make the logic and codes better to maintain I think.

Reply:
`set-owner` doesn't align with that advice as it's inconsistent with the verbs used for different types of objects. In this case, more sub-commands means that the code needs to be duplicated for every entity rather than working off a single flag. Having a single flag means it will be future-proof as well as if new entities are added as no new code is needed, while a command will mean new code will need to be added for each new entiry added.

Reply:
Given the large amount of commands and options AWS CLI has, I think they took a different approach, as a user is unlikely to be able to remember all of the commands. However, having a smaller set of commands makes it easier for users to remember them, which is where I think that advice is coming from. 'Miller's Law' states that people can only hold seven plus or minus two items in their short-term memory at any one time.

Reply:
> `set-owner` doesn't align with that advice as it's inconsistent with the verbs used for different types of objects. i.e. you are using different verbs for different objects. Also, in this case, more sub-commands mean that the code needs to be duplicated for every entity rather than working off a single flag. Having a single flag means it will be future-proof, as if new entities are added as no new code is needed, while using a command will mean new code will need to be added for each new entity added.

Use sub-command can also avoid duplicating code in every entity, that is not the key point, it is similar with how you dealing with the flag now. E.g, putting the `if("set-owner".equals(command)) {...}` in the same location of current checking for "--owner" flag. It is the same. 

Anyway, Justin I see your point, you perfer to using flags to checking user's purpose, instead of using commands. I prefer to using more commands, because in each command line, there is only one command be allowed, but multiple flags can be set. So if using more flags, the logic of checking flags will be complex and easy to cause problem; If multiple flags appeared in the same time, you have to carefully decide which one can be executed and which one won't, and you'd better give some warning message to the user.  

Here I can give an example, in the current version, with the "gcli tag set" command, user can use it to 1) set a property to the tag (with "--tag", "--property" and "--value"), and 2) set the tag to an entity (with "--name" and "--tag"). The logic is here: https://github.com/apache/gravitino/blob/main/clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java#L452

Actually, this logic has a bug. The IDE reports this error, before that, you and me didn't aware that at all:

<img width="1037" alt="image" src="https://github.com/user-attachments/assets/95a2dc4c-4415-4a65-8e9c-cb0989c2a390">

That's why I suggested to use "set-tag" command instead of combining it together with the "set" command when reviewing the previous PR. Now comes the setting "owner" function, that makes me rethink it. Besides, I don't think "set-tag" or "set-owner" command would make user hard to remember. It is quite simple word.

Reply:
I think @shaofengshi 's comment here is quite reasonable, we should always think about how to make the cli easy to use and understand, not the so-called law or principal. In the meantime we should avoid the potential ambiguity and corner-case problems.

Reply:
Thanks for pointing out that bug, I'll get that fixed.

Re code effort if we set the owner via a command, we need <no of entities> blocks of code, if we set it via a flag, it's one block of code, but let us ignore that for a moment.

Fewer commands benefit users, as people can only remember a limited number of things. Introducing additional commands may make it harder for users to keep track, and even confusing if they are similarly named, even if individual commands are relatively easy to recall. It has other benefits as well, reducing cognitive load and the learning curve, giving consistency and predictable patterns leading to a better user experience. Flags are easier to remember as you only have a small set of flags that are used with any given command.

However overloading flags can make it harder for a user to understand or predict a CLIs behavior, and flags can be less discoverable or hard to match with the command they apply to without documentation.

In general new commands are used for entirely distinct operations and flags are used modify or refine the behavior of existing commands.

There are downsides to both approaches here, and I don't think there is an ideal solution. So we need to pick one, and I think on balance using a flag makes it clearer for the user given that the operations are all similar.

Reply:
Using `set-tag` also makes the tag command rather verbose with tag being mentioned 3 times:
`gcli tag set-tag --name postgres --tag tagB`

Reply:
<img width="799" alt="image" src="https://github.com/user-attachments/assets/38b7333a-a75f-4124-88e8-3e3794760e11">
Justin, your last commit is to fix the bug in tag setting; there is another similar bug, please resolve it together. (If your IDE doesn't have such warning, you can try IntelliJ IDEA)

Reply:
I was thinking about the set action, what is the subject for "set" action? For example, you choose the subject "catalog" for set owner action `gcli catalog set --owner --name xxx --user yyy`. But for the set tag action, you choose "tag" subject for set as you mentioned above, shall we also use "catalog" subject?

Basically, we can have two choices:

1)

```
gcli catalog set --owner --name xxx --user yyy
gcli catalog set --tag --name xxx --tag-list aa,bb,cc
```

2)

```
gcli owner set --name xxx --user yyy
gcli tag set -- name xxx --tag-list aa,bb,cc
```

First thing is that we should be consistent for all the set action, second thing is that we should choose a easy-to-use way.

Reply:
For `gcli tag set --tag tagA --name postgres` you are setting a tag on the postgres entity, and the command almost reads like an English sentence.  That sentence is an imperative sentence, so the subject is implicitly you. The disadvantage with the form `gcli catalog set --tag ...` is that you explicitly need to set the entity type, which is not really needed. Why make the user type more and have to think about the entity type as well?

I initially considered doing the same with the owner command, but owner is not really an entity on its own like a metalake or tag. It's more of an attribute of an entity, which is why I didn't put the commands in the same form.

Reply:
> I was thinking about the set action, what is the subject for "set" action? For example, you choose the subject "catalog" for set owner action `gcli catalog set --owner --name xxx --user yyy`. But for the set tag action, you choose "tag" subject for set as you mentioned above, shall we also use "catalog" subject?
> 
> Basically, we can have two choices:
> 
> 
> ```
> gcli catalog set --owner --name xxx --user yyy
> gcli catalog set --tag --name xxx --tag-list aa,bb,cc
> ```
> 
> 
> ```
> gcli owner set --name xxx --user yyy
> gcli tag set -- name xxx --tag-list aa,bb,cc
> ```
> 
> First thing is that we should be consistent for all the set action, second thing is that we should choose a easy-to-use way.

I think the first choice is better; this is similar as how user performs the tagging in the UI: select the catalog/schema/table first, and then pick the tag to set, instead of the other direction. @justinmclean Justin, what do you think?

Reply:
See my comment above. I believe the second choice is better as the first requires the user to think about and specify the entity type and adds to cognative load. @jerryshao also don't want extra flags.